### PR TITLE
Profile update

### LINKED ISSUE #12: Profile MSVC: Improvements [uid, middlewares]
improvements to be made to profile as discussed:

1. Security Issue - exposing email in API endpoint; to be changed to unique id
2. Middlewares - standardising error handling and response handling

### PR DESCRIPTION
**Pull request by ZhiLinR**
updated profile microservice to include changes to the following (as per meeting)

1. User uid to use a separate field, autogenerated using ObjectId() method, sample in readme
2. tried out response and error handling middlewares + abit of standardising response formatting

also fixed a behavioural lapse in one of the endpoints - POST api/profile (register new user) - currently sensitive user data and profiling data are stored separately in collections, original post request did not trigger a record insertion into the profiling data collection

```
// user sensitive data
{
  "_id": {
    "$oid": "666350518e5c4522aed85893"
  },
  "uid": "666350518e5c4522aed85892",
  "email": "sammyho@example.com",
  "name": "Sammy"
}
```

```
// larger, more generic profiling data?
{
  "_id": {
    "$oid": "666350518e5c4522aed85894"
  },
  "uid": "666350518e5c4522aed85892",
  "lastModified": {
    "$date": "2024-06-07T18:26:12.711Z"
  },
  "gender": "female",
  "organisation_type": "banking",
  "title": "CFO"
}
```

### CODE REVIEW

File: microsvcs/user/README.md [hunk 1]
```diff
@@ -14,14 +27,36 @@ Updates specific **profiling** data (ie. organisation type, position etc) for a
     title: "CFO"
 }
 ```
-*Sample fields in the database
+Sample Success Response:
+```
+{
+    "success": true,
+    "message": {
```
Comment (cre8tion) [inline microsvcs/user/README.md]: Dont return mongodb response as api response, just return your own custom response if needed

File: microsvcs/user/app/controller/profile_con.js [hunk 2]
```diff
@@ -7,33 +8,30 @@ const QUERIES = require("../model/profile.js")
  * 1 request parameter expected.
  * @param {String} uid - unique uid field
  */
-exports.getProfileByUID = async (req, res) => {
+exports.getProfileByUID = async (req, res, next) => {
```
Comment (cre8tion) [inline microsvcs/user/app/controller/profile_con.js:11]: I believe next is only for error handling middleware, please verify on that

File: microsvcs/user/app/controller/profile_con.js [hunk 3]
```diff
@@ -7,33 +8,30 @@ const QUERIES = require("../model/profile.js")
...
-        }
-    } catch (error) {
-        if (error.message == "Server Error Occurred") {
-            res.status(500).send({ message: error.message });
+            next(UTIL.formatRes(true, result))
         } else {
-            res.status(404).send({ message: error.message });
+            throw new Error("No Reference Found")
         }
+    } catch (error) {
```
Comment (cre8tion) [inline microsvcs/user/app/controller/profile_con.js:37]: I think the error handling is supposed to catch automatically without you calling next

File: microsvcs/user/app/controller/profile_con.js (hunk 2 above)
Comment (cre8tion) [reply microsvcs/user/app/controller/profile_con.js:11]: okay seems like you have two middleware, maybe thats why. But try to verify on that

**Review by cre8tion — COMMENTED** [review]
Looks okay, except im not very sure on the middleware portion

**Comment by ZhiLinR** [comment]
changes:
1. removed MongoDB API Response
2. moved constants out of try blocks
3. standardised usage of next() in catch blocks as well
4. updated readme

**Comment by cre8tion** [comment]
LGTM, feel free to merge

**Pull request merged**